pkg/util/common.go

```diff
@@ -1347,22 +1304,31 @@ func DeleteCustomResourceDefinition(apiExtensionClient *apiextensionsclient.Clie
 }
 
 // WaitUntilPodsAreReady will wait for the pods to be ready
-func WaitUntilPodsAreReady(clientset *kubernetes.Clientset, namespace string, labelSelector string, timeoutInSeconds int64) (bool, error) {
+func WaitUntilPodsAreReady(clientset *kubernetes.Clientset, namespace string, labelSelector string, timeoutInSeconds int64) error {
+
+	// check before the first tick
+	ready, err := IsPodReady(clientset, namespace, labelSelector)
+	if err != nil || ready {
+		return err
+	}
+
 	// timer starts the timer for timeoutInSeconds. If the task doesn't completed, return error
 	timeout := time.NewTimer(time.Duration(timeoutInSeconds) * time.Second)
 	// ticker starts and execute the task for every n intervals
 	ticker := time.NewTicker(10 * time.Second)
+
+	defer ticker.Stop()
+	defer timeout.Stop()
+
 	for {
 		select {
 		case <-timeout.C:
-			ticker.Stop()
-			return false, fmt.Errorf("[NS: %s | Label: %s] the pods weren't ready - timing out after %d seconds", namespace, labelSelector, timeoutInSeconds)
+			return fmt.Errorf("[NS: %s | Label: %s] the pods weren't ready - timing out after %d seconds", namespace, labelSelector, timeoutInSeconds)
```

Comment:
IMO at timeout we should do a final check whether pods are running and return error if not.  This would account for a timeout of say 45 sec, but a ticker of 10s.  The pods could come up in the 5 seconds after the last tick check.

Reply:
done